Library/Homebrew/cmd/readall.rb

```diff
@@ -13,16 +13,8 @@ module Homebrew
 
   def readall
     if ARGV.include?("--syntax")
-      ruby_files = []
-      scan_files = %W[
-        #{HOMEBREW_LIBRARY}/*.rb
-        #{HOMEBREW_LIBRARY}/Homebrew/**/*.rb
-      ]
-      Dir.glob(scan_files).each do |rb|
-        next if rb.include?("/vendor/")
-        next if rb.include?("/cask/")
-        ruby_files << rb
-      end
+      scan_files = "#{HOMEBREW_LIBRARY}/Homebrew/**/*.rb"
+      ruby_files = Dir.glob(scan_files).reject{|file| file =~ /vendor|cask/ }
```

Comment:
`/vendor|cask/` isn't quite what there is above; should be a a preceding and trailing `/`

Reply:
@MikeMcQuaid I'm unclear about the above feedback.  I tried the following and it seemed to produce the expected results:

    files = ["foo.rb", "bar.rb", "vendor.rb", "cask.rb", "vendorcask.rb"]
    files.reject{|file| file =~ /vendor|cask/ }
    => ["foo.rb", "bar.rb"]

Reply:
Also, why is `cask` even excluded?

Reply:
@richie-thomas Sorry, I meant just including `/` inside the string e.g.
```ruby
file =~ %r{/(vendor|cask)/}
```

@reitermarkus My guess is it used to fail. @richie-thomas if you can remove `cask` and it still passes you can replace this with `file.include?("/vendor/")`

Reply:
Fixed

Reply:
Changing over to `file.include? "vendor"` causes the following test to fail:

      1) brew readall imports all Formulae for a given Tap
         Failure/Error:
           expect { brew "readall", "--aliases", "--syntax" }
             .to be_a_success
             .and not_to_output.to_stdout
             .and not_to_output.to_stderr
     
              expected #<Proc:0x000001044a3260@/usr/local/Homebrew/Library/Homebrew/test/cmd/readall_spec.rb:10> to be a success
     
       ...and:
     
          expected block to not output to stderr, but output "/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/list.rb:9: warning: assigned but unused variab...syntax error, unexpected =>, expecting end-of-input\ncask => 'invalid-header-format' do\n       ^\n"
     # ./test/cmd/readall_spec.rb:10:in `block (2 levels) in <top (required)>'
     # ./test/support/helper/spec/shared_context/integration_test.rb:44:in `block (2 levels) in <top (required)>'
     # ./test/spec_helper.rb:96:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'
    .
    Finished in 34.9 seconds (files took 1.96 seconds to load)
    198 examples, 1 failure

    Failed examples:

    rspec ./test/cmd/readall_spec.rb:2 # brew readall imports all Formulae for a given Tap

The unused variable in question, from `list.rb:9`, is the following LoC:

    one = true # rubocop:disable Lint/UselessAssignment

I didn't touch this file, so I'm not sure why removing `cask` would cause this test to fail when it didn't before.